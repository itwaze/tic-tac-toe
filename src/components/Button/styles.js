import styled from "styled-components";

export const Button = styled.button`
  display: block;
  outline: none;
  background: transparent;
  border-radius: 0.4em;
  border: 0.1em solid #ccc;
  padding: 0.5em 1em;
  transition: 0.3s;
  min-height: 3em;
  :not(:disabled) {
    cursor: pointer;
  }
`;
