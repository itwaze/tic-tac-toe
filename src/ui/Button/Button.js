import React from "react";
import styled, { css } from "styled-components";

const StyledButton = styled.button`
	display: block;
	outline: none;
	background: transparent;
	border-radius: 4px;
	border: 1px solid #ccc;
	padding: 8px 16px;
	transition: 0.3s;
	margin: ${({ margin }) => margin};
	${({ disabled }) =>
		!disabled &&
		css`
			&:hover {
				background: #eee;
				cursor: pointer;
			}
		`}
`;

const Button = (props) => {
	const { children, ...rest } = props;
	return <StyledButton {...rest}>{children}</StyledButton>;
};

export default Button;
