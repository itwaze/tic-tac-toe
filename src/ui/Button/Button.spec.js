import React from "react";
import { render } from "@testing-library/react";

import Button from "./Button";

describe("Button", () => {
  it("should take a snapshot", () => {
    const { asFragment } = render(<Button>test text</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
