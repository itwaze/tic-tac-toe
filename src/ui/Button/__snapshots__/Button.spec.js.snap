// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ gQZXKL"
  >
    test text
  </button>
</DocumentFragment>
`;
